app-name/src/RenderEntry.jsx

```diff
@@ -0,0 +1,32 @@
+import { data } from './data';
+import { FaPencilAlt } from 'react-icons/fa';
+
+export default function RenderEntry({ viewPage, onClick }) {
+  data.view = viewPage;
+
+  function handleEdit(e) {
+    onClick('create');
+    for (let i = 0; i < data.entries.length; i++) {
```

Comment:
This loop isn't doing anything, but it looks like you were hoping to eventually use this to figure out which entry was clicked. You would do that by passing the entry (or entryId) to `handleEdit` inside the `.map` code instead of here.